File: app/src/main/java/com/example/recyclerview/PostsAdapter.java
package com.example.recyclerview;

import android.view.LayoutInflater;
import android.view.ViewGroup;
import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;
import com.example.recyclerview.databinding.ItemBinding;
import java.util.List;

public class PostsAdapter extends RecyclerView.Adapter<PostsAdapter.ViewHolder> {


    public List<Post> posts;

    public void setPosts(List<Post> posts) {
        this.posts = posts;
    }


    @NonNull
    @Override
    public PostsAdapter.ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(ItemBinding.inflate(LayoutInflater.from(parent.getContext()),
                parent, false));
    }

    @Override
    public void onBindViewHolder(@NonNull PostsAdapter.ViewHolder holder, final int position) {

        Post post = posts.get(position);
        holder.binding.personNameTv.setText(post.getUserName());
        holder.binding.dateTv.setText(post.getDate());
        holder.binding.postTv.setText(post.getContent());
        holder.binding.postMv.setImageResource(post.getImage());


    }

    @Override
    public int getItemCount() {
        return posts == null ? 0 : posts.size();
    }


    public static class ViewHolder extends RecyclerView.ViewHolder {
        private ItemBinding binding;

        public ViewHolder(@NonNull ItemBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

}

File: app/src/main/java/com/example/facebook_posts/PostsAdapter.kt
package com.example.facebook_posts

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.google.android.material.behavior.HideBottomViewOnScrollBehavior
import org.w3c.dom.Text

class PostsAdapter(val items: List<Post>) : RecyclerView.Adapter<PostsAdapter.ViewHolder>() {

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val contentTextView : TextView = itemView.findViewById(R.id.post_tv)
        val nameTextView : TextView = itemView.findViewById(R.id.person_name_tv)
        val imagePostImageView : ImageView = itemView.findViewById(R.id.post_mv)
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder = ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.item, parent, false))

    override fun getItemCount(): Int = items.size

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val post = items[position]
        holder.contentTextView.text = post.content
        holder.nameTextView.text = post.name
        holder.imagePostImageView.setImageResource(post.image)

    }
}